pkg/kubelet/cm/qos_container_manager_linux_test.go

```diff
@@ -153,3 +156,37 @@ func TestQoSContainerCgroup(t *testing.T) {
 	assert.Equal(t, qosConfigs[v1.PodQOSGuaranteed].ResourceParameters.Unified["memory.min"], strconv.FormatInt(burstableMin.Value()+guaranteedMin.Value(), 10))
 	assert.Equal(t, qosConfigs[v1.PodQOSBurstable].ResourceParameters.Unified["memory.min"], strconv.FormatInt(burstableMin.Value(), 10))
 }
+
+func TestQoSContainerUpdatesOnlyOnce(t *testing.T) {
```

Comment:
Given this test I think you should look at https://pkg.go.dev/golang.org/x/sync/singleflight#Group.Do and see if you can just call that.

Reply:
Thanks, didn't know about it. The semantics are a bit different that what I am trying to achieve. For Group.Do pending requests will wait and return the result for the current invocation. In our case we want to wait for the existing invocation if there is any to finish and then wait for a new invocation that takes into account our pod to start and return the result associated with that.